File: src/dataset/typescript/nucleotide-count_exercism-samajammin/gemini/GeminiFlash_nucleotide-count_v1.ts
class NucleotideCount {
  static nucleotideCounts(dna: string) {
    const nucleotideCount: { [key: string]: number } = {
      A: 0,
      G: 0,
      T: 0,
      C: 0,
    };

    for (let i = 0; i < dna.length; i++) {
      const nucleotide = dna[i];
      if (nucleotideCount.hasOwnProperty(nucleotide)) {
        nucleotideCount[nucleotide]++;
      } else {
        throw new Error('Invalid nucleotide in strand');
      }
    }

    return nucleotideCount;
  }
}

export default NucleotideCount;

File: src/dataset/cpp/nucleotide-count_exercism-kchoadley/gemini/GeminiFlash_nucleotide_count_v1.cpp
#include "nucleotide_count.h"
#include <stdexcept>
#include <algorithm>

namespace dna {

    counter::counter(std::string strand) : nucleotides({{'A', 0}, {'C', 0}, {'G', 0}, {'T', 0}}) {
        for (char c : strand) {
            if (nucleotides.find(c) == nucleotides.end()) {
                throw std::invalid_argument("Invalid argument");
            }
            nucleotides[c]++;
        }
    }

    int counter::count(const char& c) const {
        if (nucleotides.find(c) == nucleotides.end()) {
            throw std::invalid_argument("Invalid argument");
        }
        return nucleotides.at(c);
    }

    std::map<char, int> counter::nucleotide_counts() const {
        return nucleotides;
    }

}